### ISSUE TITLE
grid.search() during an active search causes script error[Reproducer: https://jsfiddle.net/zuvk0o6w/3/]

### ISSUE THREAD

**Issue by NoelSMartin** [vitmalina/w2ui#2266]
**Short description**
Step 1: Load fiddle https://jsfiddle.net/zuvk0o6w/3/
Step 2: Start a search and matched records are highlighted
Step 3. Issue a w2ui.grid.search() statement in the console

Error:
w2ui.min.js:282 Uncaught TypeError: Cannot read properties of undefined (reading '_w2field')
    at w2grid.search (w2ui.min.js:282:66239)
    at HTMLElement.onclick ((index):1:22)

This only happens if the searches property is defined for the grid. 

**What is current behavior**
...

**What is desired behavior**
...

**Link to jsfidle/codepan with sample code**
https://jsfiddle.net/zuvk0o6w/3/

**Steps to reproduce or sample** 
Step 1: Load fiddle https://jsfiddle.net/zuvk0o6w/3/
Step 2: Start a search and matched records are highlighted
Step 3. Issue a w2ui.grid.search() statement in the console
...

**Comment by vitmalina** [comment]
I tried to repeat your steps and cannot see what the problem is. I do not see any messages in the console. The search gets applied. I tried just typing in the search input field, or clicking on >> advanced search button - all seem to work just fine. Can you rephrase what the problem is?

**Comment by NoelSMartin** [comment]
After you have applied a search by typing in the search input field.
Go to the console and try clearing the search by issues w2ui.grid.search()
You will see the error

**Comment by NoelSMartin** [comment]
```
Uncaught TypeError: Cannot read properties of undefined (reading '_w2field')
    at w2grid.search (w2ui.min.js:282:66038)
    at <anonymous>:1:11
```
![image](https://user-images.githubusercontent.com/79912980/184564018-cb314426-6686-4139-9573-438ff36f5eae.png)

**Comment by NoelSMartin** [comment]
Hello,
This is the error received

Uncaught TypeError: Cannot read properties of undefined (reading '_w2field')
    at w2grid.search (w2ui.min.js:282:66038)
    at <an

When you try to do this[image: image.png]

Thanks for looking into it.

**Comment by vitmalina** [comment]
According to doc, calling `grid.search()` w/o params should do nothing. So, I fixed the error. It just repeats sam search, basically doing `grid.search(grid.searchData, grid.last.logic)`

**Comment by NoelSMartin** [comment]
Hello @vitmalina,
In the earlier build, besides the error above, calling grid.search() w/o params would clear the search input and show all records and that was a nice feature.
Could you re-purpose grid.search() w/o params to clear the grid search please, or is there a different approach we should take for this.

I have an an implementation such as the one below using the earlier w2ui build and simply calling grid.search w/o params clears the current search
![image](https://user-images.githubusercontent.com/79912980/186794969-e0a34a07-daa9-491c-ab7e-3a29d63f93cf.png)

**Comment by vitmalina** [comment]
w2ui.grid.searchReset() - will clear search result
w2ui.grid.search('') - will also clear search result

**Comment by NoelSMartin** [comment]
> w2ui.grid.searchReset() - will clear search result w2ui.grid.search('') - will also clear search result

Thank you
